library/src/main/scala/scala/scalajs/runtime/Semantics.scala

```diff
@@ -0,0 +1,40 @@
+package scala.scalajs.runtime
+
+import scala.annotation.switch
+
+/** Utilities to test for erroneous conditions depending on the Semantics
+ *  configuration.
+ */
+object Semantics {
+
+  // Must be in sync with checkedBehaviorToInt() in OptimizerCore
+  private final val Compliant = 0
+  private final val Fatal = 1
+  private final val Unchecked = 2
+
+  // Note: can return `Unchecked` if caught as an intrinsic by the optimizer
+  private def asInstanceOfs: Int =
+    if (environmentInfo.semantics.compliantAsInstanceOfs) Compliant
+    else Fatal
```

Comment:
It only changes the behavior in an optimizing way. Indeed, when the intrinsic returns `Unchecked`, it is *also* valid to return `Compliant` or `Fatal`, since, by definition, `Unchecked` means that we can do *anything*. It's just not *optimal* to return `Fatal`, but it is still valid (insofar as how the result value of this method is used, which we control because it's private).

That was my reasoning. I can just as well put 0, 1 or 2 in the environment/linking info, and return that.